tests/test_request_attrs.py

```diff
@@ -785,6 +783,23 @@ def test_app_missing(self):
 
         assert req.app == ''
 
+    @pytest.mark.parametrize('etag,expected', [
+        ('W/"67ab43"', ['67ab43']),
+        ('w/"67ab43"', ['67ab43']),
+        ('W/"67ab43", "54ed21"', ['67ab43', '54ed21']),
+        ('  W/"67ab43" , "54ed21"', ['67ab43', '54ed21']),
+        ('W/"67ab43"  ,  "54ed21"', ['67ab43', '54ed21']),
+        ('W/"67ab43" ,"54ed21"', ['67ab43', '54ed21']),
+        ('W/"67ab43", unquoted, "54ed21"', ['67ab43', '54ed21']),
```

Comment:
Looking at bug trackers of various open source projects, or at any type of projects with public bug trackers, it is very common to find issues mentioning that they are not properly quoting ``ETag``s.

On one hand, RFCs are probably the only area where Falcon is opinionated -- that we should follow the established standards, but OTOH rejecting unquoted tags might be perceived as harsh. Could we maybe consider allowing a single unquoted ``ETag`` value in the header?

Regardless how we go forward with the unquoted case, we should IMHO document the behaviour clearly, and potentially reconsider how we handle parse errors. Right now, we would silently swallow any unrecognized tags, and thus mask the corresponding parse errors. Masking errors this way might lead to hard-to-debug scenarios.

Reply:
It isn't terribly out of the ordinary for updated RFCs to acknowledge dubious behavior that has proliferated and so must be allowed for the time being, while encouraging a stricter adherence to the preferred syntax going forward. With that in mind, I think we should probably be a little more forgiving in this specific case. I like @vytas7 suggestion:

> Could we maybe consider allowing a single unquoted ETag value in the header?